src/foam/nanos/pm/PM.js

```diff
@@ -84,7 +84,9 @@ foam.CLASS({
       javaCode: `
     if ( x == null ) return;
     if ( getIsError() ) return;
-    setEndTime(new java.util.Date());
+    if ( getEndTime() == null ) {
+      setEndTime(new java.util.Date());
```

Comment:
This reminds me that using a Date is much more expensive that using a long timestamp in millieseconds, BTW.